releasenotes/notes/remove-payload-submodule-58fac4aaf3d4f5da.yaml

```
@@ -0,0 +1,4 @@
+---
+other:
+  - |
+    The payload submodule has been removed.
```

nit:
```suggestion
    The `ddtrace.payload` submodule has been removed.
```

might be slightly more useful for a user reading the notes 🙂